kroxylicious/src/main/java/io/kroxylicious/proxy/internal/codec/ByteBufAccessorImpl.java

```diff
@@ -131,8 +131,10 @@ public double readDouble() {
     }
 
     @Override
-    public void readArray(byte[] arr) {
-        buf.readBytes(arr);
+    public byte[] readArray(int length) {
+        byte[] dst = new byte[length];
```

Comment:
I think we need to check that the length is less than the remaining bytes in `buf` _before_ the allocation.

Reply:
@k-wall, could you address this comment, so we can get this PR merged? How is it related to this version update btw.?

Reply:
Ok, so looking deeper I answered my own question.  You are following the pattern of the Kafka implementation.  I'll do the same.

Reply:
Done.